base/math.jl

```diff
@@ -103,20 +103,23 @@ function evalpoly(x, p::Tuple)
         _evalpoly(x, p)
     end
 end
+evalpoly(x, ::Tuple{}) = zero(one(x)) # dimensionless zero, i.e. 0 * x^0
```

Comment:
I agree that this is right. Would you please add a test that will fail under the `zero(x)` implementation?

Reply:
added a test